hack/update-bazel.sh

```diff
@@ -18,99 +18,25 @@ set -o nounset
 set -o pipefail
 
 KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
-source "${KUBE_ROOT}/hack/lib/init.sh"
 
-# Ensure that we find the binaries we build before anything else.
-export GOBIN="${KUBE_OUTPUT_BINPATH}"
-PATH="${GOBIN}:${PATH}"
+# delete all bazel related files
```

Comment:
I'll be writing a more detailed notice but the TLDR on why this change is so authors with in-flight PRs containing bazel related diffs (e.g. because they added a go source file) can just do:
- rebase to upstream after this merges
- run `hack/update-bazel.sh` (now _very_ fast) during the rebase to drop all bazel related generated files
- carry on with their PR as usual